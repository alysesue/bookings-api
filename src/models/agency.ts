import { BaseEntity, Column, Entity, Index, PrimaryGeneratedColumn } from "typeorm";

@Entity()
export class Agency extends BaseEntity {

	@PrimaryGeneratedColumn()
	private _id: number;

	@Column()
	@Index({unique: true})
	private _name: string;

	public set name(name: string) {
		this._name = name;
	}
}
