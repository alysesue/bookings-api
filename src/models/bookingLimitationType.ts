export enum BookingLimitationType {
	NoLimitations = 'NoLimitations',
	LimitedBookingPerDate = 'LimitedBookingPerDate',
	LimitedUpcomingBooking = 'LimitedUpcomingBooking',
}
