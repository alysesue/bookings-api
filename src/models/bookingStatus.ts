export enum BookingStatus {
	PendingApproval = 1,
	Accepted = 2,
	Closed = 3,
	Cancelled = 4,
}
