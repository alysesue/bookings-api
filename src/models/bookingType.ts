export enum BookingType {
	Created = 'Created',
	Updated = 'Updated',
	CancelledOrRejected = 'CancelledOrRejected',
}
