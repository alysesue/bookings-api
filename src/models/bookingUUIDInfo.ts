export type BookingUUIDInfo = {
	bookingUUID: string;
	bookingId: number;
	serviceId: number;
	organisationId: number;
	serviceProviderId?: number;
};
