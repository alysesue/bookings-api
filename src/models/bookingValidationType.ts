export enum BookingValidationType {
	Citizen = 'citizen',
	Admin = 'admin'
}
