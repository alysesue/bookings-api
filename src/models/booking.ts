import {BaseEntity, Column, Entity, PrimaryGeneratedColumn} from "typeorm";
import {BookingStatus} from "./bookingStatus";

@Entity()
export class Booking extends BaseEntity {
	@PrimaryGeneratedColumn()
	private _id: number;

	@Column()
	private _status: BookingStatus;

	@Column()
	private _startDateTime: Date;
	@Column()
	private _sessionDurationInMinutes: number;

	constructor(startDateTime: Date, sessionDurationInMinutes: number) {
		super();
		this._startDateTime = startDateTime;
		this._sessionDurationInMinutes = sessionDurationInMinutes;
		this._status = BookingStatus.PendingApproval;
	}

	public get id(): number {
		return this._id;
	}

	public set status(newStatus: BookingStatus) {
		this._status = newStatus;
	}

	public get status(): BookingStatus {
		return this._status;
	}

	public get startDateTime(): Date {
		return this._startDateTime;
	}

	public get sessionDurationInMinutes(): number {
		return this._sessionDurationInMinutes;
	}

	public getSessionEndTime(): Date {
		return new Date(this._startDateTime.getTime() + this._sessionDurationInMinutes * 60 * 1000);
	}
}
