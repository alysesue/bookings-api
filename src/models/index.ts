import { Booking } from "./booking";
import { BookingStatus } from "./bookingStatus";
import { Schedule } from "./schedule";
import { WeekDaySchedule } from "./weekDaySchedule";
import { Calendar } from "./calendar";
import { TimeOfDay } from './timeOfDay';
import { ServiceProvider } from "./serviceProvider";
import { BusinessValidation } from './businessValidation';
import { Timeslot } from '../models/timeslot';

export { Booking, BusinessValidation, BookingStatus, Schedule, WeekDaySchedule, Calendar, TimeOfDay, Timeslot, ServiceProvider };
