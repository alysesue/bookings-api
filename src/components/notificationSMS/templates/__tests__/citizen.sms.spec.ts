import { Booking, Service } from '../../../../models';
import {
	CitizenSMSTemplateBookingActionByCitizen,
	CitizenSMSTemplateBookingActionByServiceProvider,
} from '../citizen.sms';

describe('Notification SMS templates tests', () => {
	const booking = new Booking();
	booking.startDateTime = new Date(2021, 3, 14, 10);
	booking.endDateTime = new Date(2021, 3, 14, 11);
	booking.service = { name: 'Career' } as Service;
	booking.status = 1;
	booking.location = 'Some street';
	booking.serviceProviderId = 1;
	booking.videoConferenceUrl = 'http://www.zoom.us/1234567';

	it('should create citizen sms for citizen created booking', () => {
		const result = new CitizenSMSTemplateBookingActionByCitizen().CreatedBookingSMS(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create citizen sms for citizen updated booking', () => {
		const result = new CitizenSMSTemplateBookingActionByCitizen().UpdatedBookingSMS(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create citizen sms for citizen cancelled booking', () => {
		const result = new CitizenSMSTemplateBookingActionByCitizen().CancelledBookingSMS(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create citizen sms for service provider created booking', () => {
		const result = new CitizenSMSTemplateBookingActionByServiceProvider().CreatedBookingSMS(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create citizen sms for service provider updated booking', () => {
		const result = new CitizenSMSTemplateBookingActionByServiceProvider().UpdatedBookingSMS(booking);
		expect(result).toMatchSnapshot();
	});

	it('should create citizen sms for service provider cancelled booking', () => {
		const result = new CitizenSMSTemplateBookingActionByServiceProvider().CancelledBookingSMS(booking);
		expect(result).toMatchSnapshot();
	});
});
