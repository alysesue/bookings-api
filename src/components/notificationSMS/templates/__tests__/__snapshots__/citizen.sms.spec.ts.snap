// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification SMS templates tests should create citizen sms for citizen cancelled booking 1`] = `
"BookingSG cancellation: Career
		
You have cancelled the following booking.

Booking for: Career.

Booking status: Pending Approval
Date: 14 April 2021
Time: 10:00am - 11:00am
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>"
`;

exports[`Notification SMS templates tests should create citizen sms for citizen created booking 1`] = `
"
BookingSG confirmation: Career

Your booking request has been received.

Booking for: Career.

Below is a confirmation of your booking details.
Booking status: Pending Approval
Date: 14 April 2021
Time: 10:00am - 11:00am
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>"
`;

exports[`Notification SMS templates tests should create citizen sms for citizen updated booking 1`] = `
"
BookingSG update: Career

You have updated a booking.

Booking for: Career.

Below is a confirmation of your updated booking details.
Booking status: Pending Approval
Date: 14 April 2021
Time: 10:00am - 11:00am
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>"
`;

exports[`Notification SMS templates tests should create citizen sms for service provider cancelled booking 1`] = `
"BookingSG cancellation: Career
		
The following booking has been cancelled by the other party.

Booking for: Career.

Booking status: Pending Approval
Date: 14 April 2021
Time: 10:00am - 11:00am
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>"
`;

exports[`Notification SMS templates tests should create citizen sms for service provider created booking 1`] = `
"BookingSG confirmation: Career

A booking has been made.

Booking for: Career.

Below is a confirmation of your booking details.
Booking status: Pending Approval
Date: 14 April 2021
Time: 10:00am - 11:00am
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>
"
`;

exports[`Notification SMS templates tests should create citizen sms for service provider updated booking 1`] = `
"
			BookingSG update: Career
			
There has been an update to your booking confirmation.

Booking for: Career.

Below is a confirmation of your updated booking details.
Booking status: Pending Approval
Date: 14 April 2021
Time: 10:00am - 11:00am
Video Conference Link: <a href='http://www.zoom.us/1234567'>http://www.zoom.us/1234567</a>
Location: <b>Some street</b>"
`;
